python/cudf/cudf/_config/config.py

```diff
@@ -0,0 +1,30 @@
+from contextlib import contextmanager
+
+_OPTIONS = {
+    # nulls compare like NaN/NaT in numeric/datetime comparisons
+    "nulls_compare_like_nans": False
+}
+
+
+@contextmanager
+def option_context(*args):
```

Comment:
Is this implementation thread-safe?

Reply:
Yes, since no two threads can be modifying Python objects at the same time, by virtue of the GIL

Reply:
You're right -- this is not thread safe. In an offline conversation, you made the very good point that, within this context, one can release the GIL, and another thread can then pick up the wrong option.